File: StarcevMatvey/Term 2/Block 2/Task_1/Bots/RandomBot.cs
using BlackJack;

namespace Bots
{
    public class RandomBot : Player
    {
        private Random random;

        public RandomBot(string name, int balance) : base(name, balance)
        {
            random = new Random();
        }

        public override int GetNewBet()
        {
            return random.Next(Balance);
        }

        public override string GetMove()
        {
            string[] possibleOptions = { "hit", "double", "stand" };
            return possibleOptions[random.Next(possibleOptions.Length)];
        }

    }
}

File: IvanShurenkov/Second Term/Block2/Task1/src/main/kotlin/lib/blackjack/bots/RandomBot.kt
package lib.blackjack.bots

import lib.blackjack.base.GameInfo
import lib.blackjack.base.Hand
import lib.blackjack.base.IPlayer
import lib.blackjack.base.PlayerMove
import lib.blackjack.base.PlayerMove.HIT
import lib.blackjack.base.PlayerMove.STAND
import kotlin.random.Random.Default.nextBoolean

object RandomBot : IPlayer {
    override val name: String = "RandomBot"

    override fun getBet(bankroll: UInt, gameInfo: GameInfo): UInt {
        return gameInfo.rangeBet.first
    }

    override fun getMove(hand: Hand, gameInfo: GameInfo): PlayerMove {
        return when {
            nextBoolean() -> HIT
            else -> STAND
        }
    }
}